### DISCUSSION TITLE
Some problem reading JSON Lines input

### DISCUSSION THREAD

**Discussion by aborruso** [saulpw/visidata#1208] [Q&A] [answered]
When I open this JSON Lines file

```
{"data": "2020-02-24T18:00:00", "stato": "ITA", "codice_regione": 18, "denominazione_regione": "Calabria", "lat": 38.90597598, "long": 16.59440194, "ricoverati_con_sintomi": 0, "terapia_intensiva": 0, "totale_ospedalizzati": 0, "isolamento_domiciliare": 0, "totale_positivi": 0, "variazione_totale_positivi": 0, "nuovi_positivi": 0, "dimessi_guariti": 0, "deceduti": 0, "casi_da_sospetto_diagnostico": "", "casi_da_screening": "", "totale_casi": 0, "tamponi": 1, "casi_testati": "", "note": "", "ingressi_terapia_intensiva": "", "note_test": "", "note_casi": "", "totale_positivi_test_molecolare": "", "totale_positivi_test_antigenico_rapido": "", "tamponi_test_molecolare": "", "tamponi_test_antigenico_rapido": "", "codice_nuts_1": "", "codice_nuts_2": ""}
{"data": "2020-02-24T18:00:00", "stato": "ITA", "codice_regione": 15, "denominazione_regione": "Campania", "lat": 40.83956555, "long": 14.25084984, "ricoverati_con_sintomi": 0, "terapia_intensiva": 0, "totale_ospedalizzati": 0, "isolamento_domiciliare": 0, "totale_positivi": 0, "variazione_totale_positivi": 0, "nuovi_positivi": 0, "dimessi_guariti": 0, "deceduti": 0, "casi_da_sospetto_diagnostico": "", "casi_da_screening": "", "totale_casi": 0, "tamponi": 10, "casi_testati": "", "note": "", "ingressi_terapia_intensiva": "", "note_test": "", "note_casi": "", "totale_positivi_test_molecolare": "", "totale_positivi_test_antigenico_rapido": "", "tamponi_test_molecolare": "", "tamponi_test_antigenico_rapido": "", "codice_nuts_1": "", "codice_nuts_2": ""}
{"data": "2020-02-24T18:00:00", "stato": "ITA", "codice_regione": 08, "denominazione_regione": "Emilia-Romagna", "lat": 44.49436681, "long": 11.341720800000001, "ricoverati_con_sintomi": 10, "terapia_intensiva": 2, "totale_ospedalizzati": 12, "isolamento_domiciliare": 6, "totale_positivi": 18, "variazione_totale_positivi": 0, "nuovi_positivi": 18, "dimessi_guariti": 0, "deceduti": 0, "casi_da_sospetto_diagnostico": "", "casi_da_screening": "", "totale_casi": 18, "tamponi": 148, "casi_testati": "", "note": "", "ingressi_terapia_intensiva": "", "note_test": "", "note_casi": "", "totale_positivi_test_molecolare": "", "totale_positivi_test_antigenico_rapido": "", "tamponi_test_molecolare": "", "tamponi_test_antigenico_rapido": "", "codice_nuts_1": "", "codice_nuts_2": ""}
```

vd reads properly the first 2 lines. I do not understand why it does not read properly the last one

![image](https://user-images.githubusercontent.com/30607/142737862-8bdb8783-e4d0-4e1b-92b0-4d8b39a50578.png)

I'm using `saul.pw/VisiData v2.8dev`.

Thank you

**Comment by geekscrapy** [comment, marked as answer]
Hmm doesn't seem to be visidata specific (from https://codverter.com/src/jsonvalidator):

![image](https://user-images.githubusercontent.com/11225502/142738365-eadbca9b-fc74-49fc-b81a-230d45d5ff53.png)

**Comment by geekscrapy** [comment]
This might be the same in python: https://stackoverflow.com/questions/11786758/why-parseint08-is-giving-0-whereas-parseint07-is-giving-7

[update]
Seems like it
![image](https://user-images.githubusercontent.com/11225502/142738557-08b5758a-d4fe-452b-9665-133b150dabdb.png)

https://stackoverflow.com/questions/27361565/why-is-json-invalid-if-an-integer-begins-with-a-leading-zero#:~:text=A%20leading%200%20indicates%20an,would%20not%20contain%20an%208.

**Comment by saulpw** [comment]
JSON (and many other languages) parses numbers with a leading `0` as octal, meaning that only digits 0-7 are allowed (and it's parsed in base 8, which may yield surprising results:  `031` will be parsed as 25).  So it makes sense that this generates an error.  (Although VisiData should continue parsing later rows and only show errors for the problem row). The more interesting question is, how did you come across this JSON?  Whatever is generating it is printing these numbers incorrectly for some reason; they should probably be strings.

**Comment by aborruso** [comment]
Thanks, it was very simple, but I didn't see it.

I don't know how it was generated. I ask my friend who sent it to me.
Thanks

**Comment by saulpw** [comment]
Also remember that you can use `z Ctrl+E` in one of the error cells, and it should show a message like in @geekscrapy 's screenshot.

**Comment by aborruso** [comment]
> Hmm doesn't seem to be visidata specific (from https://codverter.com/src/jsonvalidator):

Before writing here I had done a test with jq (`<input.jsonl jq . -c`). It did not return any errors to the screen and so I wrote here. Excuse me again

**Reply by anjakefala** [reply]
Don't worry @aborruso! This can be a space to help debug, too. =)
